src/org/omegat/core/data/RealProject.java

```diff
@@ -1857,7 +1865,7 @@ public void addTranslation(String id, String source, String translation, boolean
                     }
                     tr.source = sourceS;
                     tr.translation = transS;
-                    data.put(sourceS, new TMXEntry(tr, true, null));
+                    data.put(new EntryKey(sourcePath, sourceS, id, "", "", ""), new TMXEntry(tr, true, null));
```

Comment:
If at line 1817/1818 you change to ITMXEntry, no need to call a constructor:
data.put(new EntryKey(sourcePath, sourceS, id, "", "", ""), tr);